velox/functions/prestosql/Reduce.cpp

```diff
@@ -81,13 +81,14 @@ class ReduceFunction : public exec::VectorFunction {
     const auto& initialState = args[1];
     auto partialResult =
         BaseVector::create(initialState->type(), rows.end(), context.pool());
-
+    std::vector<bool> isNullArrays(rows.end(), false);
     // Process null and empty arrays.
     auto* rawNulls = flatArray->rawNulls();
     auto* rawSizes = flatArray->rawSizes();
     rows.applyToSelected([&](auto row) {
       if (rawNulls && bits::isBitNull(rawNulls, row)) {
         partialResult->setNull(row, true);
+        isNullArrays[row] = true;
```

Comment:
Partial result is set as null for this row, do we still need `isNullArrays` to record the nullability?

Reply:
Partial result is set as null but it can't distinguish Null Array and Array with null element in later functions, thus cause the the result is same for these 2 cases and not correct. So we make a special marker for Null Array and apply this maker on the final result.

Reply:
> it can't distinguish Null Array and Array with null element.

Is it the expected behavior by setting null for both null array and array with null element?

Reply:
In this line, `partialResult` is only setting null true for NULL Array. For array with null element, the `setNull` should be done during merge function process. The problem is that even when `partialResult` is setting null true there, this value will be erased during finish function like `coalesce(s, 0)`, which will change `null` value to `0`, but it shouldn't apply on NULL array.

Reply:
Is above the expected behavior of Spark? I noticed this function is a presto one, and are they the same?

Reply:
This is expected behavior of Spark. But no matter for Spark or Presto, NULL array handle should be the same.